src/Styleguide/Elements/RadioGroup.tsx

```diff
@@ -1,25 +1,20 @@
-import { color, space } from "@artsy/palette"
 import React from "react"
 import { BorderProps, SizeProps, SpaceProps } from "styled-system"
 
-import { css } from "styled-components"
+import { BorderBox } from "Styleguide/Elements/Box"
 import { Flex } from "Styleguide/Elements/Flex"
-import { Radio } from "Styleguide/Elements/Radio"
+import { Join } from "Styleguide/Elements/Join"
+import { Separator } from "Styleguide/Elements/Separator"
+
+export interface RadioGroupElementProps {
+  value?: string
+}
 
 export interface RadioGroupProps {
   disabled?: boolean
-  onSelect: (selectedOption: string) => void
+  onSelect?: (selectedOption: string) => void
   defaultValue?: string
-  options: Array<{ label: React.ReactNode | null; id: string }>
-  renderRadio?: (
-    props: {
-      id: string
-      label: React.ReactNode
-      selected: boolean
-      onSelect: () => void
-      disabled: boolean
-    }
-  ) => React.ReactNode | null
+  children: Array<React.ReactElement<RadioGroupElementProps>>
```

Comment:
I added this type to make it slightly safer in terms of the `children` usage. Of course it wouldn't make it perfectly safe but at least we can get a warning at compile time:

![screen shot 2018-08-14 at 3 10 48 pm](https://user-images.githubusercontent.com/386234/44112927-85386d30-9fd4-11e8-8eb9-0dc9dbcceb6d.png)